brainiak/isfc.py

```diff
@@ -1,226 +1,1104 @@
+                  group_labels[1]: group_assignment.count(group_labels[1])}
+
+        # For two-sample pairwise approach set up selector from matrix
+        if pairwise == True:
+            # Sort the group_assignment variable if it came in shuffled
+            # so it's easier to build group assignment matrix
+            sorter = np.array(group_assignment).argsort()
+            unsorter = np.array(group_assignment).argsort().argsort()
+            
+            # Populate a matrix with group assignments                                 
+            group_matrix = np.vstack((np.hstack((np.full((groups[group_labels[0]],
+                                                          groups[group_labels[0]]),
+                                                         group_labels[0]),
+                                                 np.full((groups[group_labels[0]],
+                                                          groups[group_labels[1]]),
+                                                        np.nan))),
+                                      np.hstack((np.full((groups[group_labels[1]],
+                                                          groups[group_labels[0]]),
+                                                         np.nan),
+                                                 np.full((groups[group_labels[1]],
+                                                          groups[group_labels[1]]),
+                                                         group_labels[1])))))
+            np.fill_diagonal(group_matrix, np.nan)
+            
+            # Unsort matrix and squareform to create selector
+            group_selector = squareform(group_matrix[unsorter, :][:, unsorter],
+                                        checks=False)
+            
+        # If leave-one-out approach, just user group assignment as selector
+        elif pairwise == False:
+            group_selector = group_assignment
+    
+    # Manage one-sample and incorrect group assignments
+    elif not group_assignment or len(np.unique(group_assignment)) == 1:
+        n_groups = 1
+        
+        # If pairwise initialize matrix of ones for sign-flipping
+        ones_matrix = np.ones((n_subjects, n_subjects))
+        
+    elif len(np.unique(group_assignment)) > 2:
+        raise ValueError("This test is not valid for more than "
+                         f"2 groups! (got {n_groups})")
+    else:
+        raise ValueError("Invalid group assignments!")
+    
+    # Infer subjects, groups, voxels and print for user to check
+    n_voxels = iscs.shape[1]
+    print(f"Assuming {n_subjects} subjects, {n_groups} group(s), "
+          f"and {n_voxels} voxel(s) or ROI(s).")
+    
+    # Set up permutation type (exact or Monte Carlo)
+    if n_groups == 1:
+        if n_permutations < 2**n_subjects:
+            print("One-sample approximate permutation test using sign-flipping "
+                  "procedure with Monte Carlo resampling.")
```

Comment:
Actually, we do not use `print` at all. We use logging instead, because it is easier to control. This is enforced by `pr-check` via `run-checks`. See the documentation for contributors:
http://brainiak.org/docs/contributing.html#standards

Reply:
Replacing any print statements with logging functionality...

Reply:
Indeed, I think the flag is not necessary. Our documentation points to the guidelines in the official Python documentation. You can use info or debug level for expected operations; here are the options for warnings:
"warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning"
"logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted"
https://docs.python.org/3/howto/logging.html#when-to-use-logging